trendmicro_deepsecurity/plugin.spec.yaml

```diff
@@ -0,0 +1,127 @@
+plugin_spec_version: v2
+extension: plugin
+products: [insightconnect]
+name: trendmicro_deepsecurity
+title: Trend Micro Deep Security
+description: Utilise the Deep Security IPS engine to protect against vulnerabilities without installing a patch (virtual patching)
+version: 1.0.0
+vendor: philipp-behmer
+support: community
+status: []
+enable_cache: false
+resources:
+  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/trendmicro_deepsecurity
+  license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://philippbehmer.de
+tags:
+- vulnerability_scanning
+- vulnerability_management
+- intrusion_detection
+- intrusion_protection
+- virtual_patching
+- remediation
+- trend_micro
+- deep_security
+hub_tags:
+  use_cases: [vulnerability_management, cloud_security, remediation_management, threat_detection_and_response ]
+  keywords: [vulnerability, scanning, management, remediation, virtual, patching, intrusion, detection, prevention, remediation, trend, micro, deep, security, trendmicro, deepsecurity]
+  features: []
+connection:
+  asset_hostname_dsm:
+    title: Asset Hostname on DSM
+    description: Hostname of the asset in the Deep Security Manager
+    type: string
+    required: false
+  dsm_url:
+    title: Deep Security Manager URL
+    description: URL of the Deep Security Manager
+    type: string
+    default: https://app.deepsecurity.trendmicro.com
+    required: true
+  dsm_api_key:
+    title: Deep Security Manager API Key
+    description: API key of the Deep Security Manager
+    type: string
+    required: true
+  asset_hostname_ivm:
+    title: Asset Hostname on IVM
+    description: Hostname of the asset in InsightVM
+    type: string
+    required: false
+  ivm_url:
+    title: InsightVM URL
+    description: URL of the InsightVM console
+    type: string
+    default: https://insightvm.company.de:3780/
+    required: false
+  ivm_user:
+    title: Username
+    description: InsightVM Username
+    type: string
+    default: nxadmin
+    required: false
+  ivm_password:
```

Comment:
You'll need to use the username_password credentials. There is an example here in connections you can use. This basically combines the username and password type you have created.
https://komand.github.io/python/sdk.html?highlight=username_password#connections

Reply:
done